l10n_fr_intrastat_service/models/intrastat_service.py

```diff
@@ -0,0 +1,378 @@
+
+                if any([
+                        line_tax.exclude_from_intrastat_if_present
+                        for line_tax in line.invoice_line_tax_ids]):
+                    continue
+
+                if line.product_id.is_accessory_cost:
+                    amount_invoice_cur_accessory_cost += line.price_subtotal
+                else:
+                    amount_invoice_cur_regular_service += line.price_subtotal
+
+            # END of the loop on invoice lines
+            if regular_product_in_invoice:
+                amount_invoice_cur_to_write = \
+                    amount_invoice_cur_regular_service
+            else:
+                amount_invoice_cur_to_write = (
+                    amount_invoice_cur_regular_service +
+                    amount_invoice_cur_accessory_cost)
+
+            if invoice.currency_id != eur_cur:
+                amount_company_cur_to_write = int(round(
+                    invoice.currency_id.with_context(
+                        date=invoice.date_invoice).compute(
+                        amount_invoice_cur_to_write,
+                        self.company_id.currency_id)))
+            else:
+                amount_company_cur_to_write = int(round(
+                    amount_invoice_cur_to_write))
+
+            if amount_company_cur_to_write:
+                if invoice.type == 'out_refund':
+                    amount_invoice_cur_to_write *= -1
+                    amount_company_cur_to_write *= -1
+
+                # Why do I check that the Partner has a VAT number
+                # only here and not earlier ? Because, if I sell
+                # to a physical person in the EU with VAT, then
+                # the corresponding partner will not have a VAT
+                # number, and the entry will be skipped because
+                # line_tax.exclude_from_intrastat_if_present is
+                # always True and amount_company_cur_to_write = 0
+                # So we should not block with a raise before the
+                # end of the loop on the invoice lines and the "if
+                # amount_company_cur_to_write:"
+                if not invoice.partner_id.vat:
+                    raise UserError(_(
+                        "Missing VAT number on partner '%s'.")
+                        % invoice.partner_id.display_name)
+                else:
+                    partner_vat_to_write = invoice.partner_id.vat
+
+                line_obj.create({
+                    'parent_id': self.id,
+                    'invoice_id': invoice.id,
+                    'partner_vat': partner_vat_to_write,
+                    'partner_id': invoice.partner_id.id,
+                    'invoice_currency_id': invoice.currency_id.id,
+                    'amount_invoice_currency': amount_invoice_cur_to_write,
+                    'amount_company_currency': amount_company_cur_to_write,
+                    })
+        self.message_post(_('Re-generating lines from invoices'))
```

Comment:
This line produces an error: `TypeError: message_post() takes 1 positional argument but 2 were given`.

```suggestion
        self.message_post(body=_('Re-generating lines from invoices'))
```

Reply:
I also get this crash. But, when I look at the proto of the method, it seems ok : https://github.com/odoo/odoo/blob/12.0/addons/mail/models/mail_thread.py#L1931
The named arg "body" is the first argument of message_post(), so it should work without "body=", no ?

Reply:
message_post() had a similar proto on odoo v10, so maybe it is a change of behavior between python2 and python3...

Reply:
@alexis-via the problem is that there are all kinds of message_post overrides, such as `def message_post(self, **kwargs):`. So this method is really intended to be used with keyword arguments only.

Actually in python3, it should better be declared like this: `def message_post(self, *, body='', ...)` to force users to call its with keyword arguments.

Reply:
https://github.com/odoo/odoo/pull/33306